File: src/1512. Number of Good Pairs.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        counts = [0] * 101
        for v in nums:
            counts[v] += 1
            
        pair = 0
        for count in counts:
            pair += count * (count - 1) // 2
        return pair

File: src/1512. Number of Good Pairs.cpp
class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        int counts[101] = {};
        for (auto v : nums) {
            counts[v]++;
        }

        int pair = 0;
        for (auto count : counts) {
            pair += count * (count - 1) / 2;
        }
        return pair;
    }
};